### ISSUE TITLE
Every time I open the .editorconfig it adds things to the file I don't want

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#59325]

**Comment by jmarolf** [comment]
whoops, thanks @Youssef1313 the bot that moves these over picks a repo based on areas path and I forgot to change this to the roslyn one before moving it to github.

**Comment by jmarolf** [comment]
related to https://github.com/dotnet/roslyn/issues/52735

**Comment by 333fred** [comment]
I hit this myself just now. I have an editorconfig where I've only specified 3 keys, related to tab and indent control for a specific folder, and every time I accidentally open the file in the editorconfig UI it adds a whole of keys that I did not have set and don't want to be set. As much as I want to like the editorconfig UI, I find myself completely unable to trust it, given the issues around inserting unwanted things and not saving correctly (https://github.com/dotnet/roslyn/issues/54556 for that one).

**Comment by majorregal** [comment]
Yes for the love of god, please fix this. EditorConfig UI is unusable with this bug and simply dropping a `.editorconfig` file in the root solution/project directory (tried both) does nothing. It seems like Visual Studio only does something when "directed" through the EditorConfig UI.

**Comment by alexdresko** [comment]
There are SO many problems with the .editorconfig UI. I think I'm just going to avoid it and edit the file manually now that I've got it somewhere close to how I want it.

**Comment by ltlombardi** [comment]
This ticket just turned 2. Happy Birthday

**Comment by VincentH-Net** [comment]
In Visual Studio 17.9.5, after changing settings in the designer and saving, duplicate entries are added in either an additional `[*.{cs,vb}]` section or an existing section other than `[*.cs]`.

The editor is both the default and unusable due to this behaviour.

**Comment by wforney** [comment]
Why do we need a GUI for this? Why don't you guys fix the intellisense for it so it actually shows the rules and ditch this crappy GUI that doesn't honor the groups and has all these issues? Who though it was a good idea to **break** the intellisense and add a crappy GUI?

**Comment by wforney** [comment]
> In Visual Studio 17.9.5, after changing settings in the designer and saving, entries that are already in the `[*.cs]` section are added in either an additional `[*.{cs,vb}]` section or an existing section other than `[*.cs]`.
> 
> The editor is both the default and unusable due to this behaviour.

You do know you can switch to the raw file with F7? But the intellisense was broken around the time they added the GUI, presumably to push people towards using it... though it is limited... I tried to use the GUI. It sucks. I just open the file, hit F7 and close the GUI and edit by hand without intellisense. Might as well be using notepad.

**Comment by stamminator** [comment]
@wforney I didn't know about the F7 trick. Thank you.

It seems to me the needed improvements to Visual Studio's EditorConfig integration are as follows:

1. Embed a "`<>` View Code" button somewhere in the .editorconfig GUI. Perhaps in the upper right corner, the same way VS Code does for the Settings GUI with an "Open Settings (JSON)" button. Needing to know the F7 trick is too hidden.
2. Implement proper syntax highlighting and intellisense in the raw .editorconfig file using [the spec implementation Visual Studio already uses](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules).
3. Fix the issue in the EditorConfig GUI where new items are added merely from opening it.
4. Nice-to-have: allow creation of new naming rules through the GUI. Currently, existing rules can be partially modified, but adding or removing requires editing the raw file.

**Comment by danbopes** [comment]
I encountered this as well. Was annoying to have to workaround with editing the file manually.

**Comment by Vlad-Stryapko** [comment]
Just hit the same issue, can confirm the GUI is actively harmful. Hopefully it gets fixed eventually, as of now the GUI is to be avoided like a plague.

**Comment by wforney** [comment]
> Just hit the same issue, can confirm the GUI is actively harmful. Hopefully it gets fixed eventually, as of now the GUI is to be avoided like a plague. 

They should just fix the intellisense so the auto complete works again. Shortly before the GUI launched they deliberately broke it. The GUI is an abomination.

**Comment by nithinbandaru1** [comment]
I really like GUI compared to intelligence. Just because it clearly shows the all available options very well organized and grouped (Can be improved).

Issues with GUI are quite annoying right now to use because of below issues.

1. Generated .editorconfig is getting modified on opening.
    1.1 Additionally severity details is getting added. Why not add them by default and also for all properties.
    1.2 `[*.{cs,vb}]` section is getting added though I am using only c# project and also I see some duplicate rules that are already present in `[*.c#]` section.
2. Editor config organizing is not reflected in editorconfig file, which is making things difficult to work with editorconfig directly.
3. Syntax `dotnet_diagnostic.<RuleId> = true:<severity>` not supported. Could give better consistency with this shorthand syntax instead of using 2 pieces separately.
4. No opening raw file option in the GUI.
5. How to manage when there is use of multiple programming language code styles.
   5.1 A drop down filter in GUI missing
   5.2. I don't even see an export option in VS Settings too. So, VS supports code styles only for csharp?
6. Is there any option to do export of editorconfig using dotnet command. Microsoft is encouraging new comers to use CLI for easy on-boarding. So CLI option matters I guess?

> - [`Easy improvement`] At-least 2, 4 and 5 can be implemented very easily I guess.
> - [`Bug`] I see 1 as a blocker. Please consider it priority 1 to make it a bit usable.
> - [`New features`] I see now 3 and 6 probably needing some work.

Hope you guys address this and make life easier for developers.

**Comment by smaillet** [comment]
The discussion here has gone far afield of the actual bug. To restate the bug, that still exists ***ALMOST 4 YEARS LATER*** 😒:  

***The editor is ALTERING a file when it is opened***.  

That's the bug and should be addressed with high priority as it is confusing and breaks usability of the editor. No editor should ever alter a file without user action.

**Comment by CyrusNajmabadi** [comment]
> No editor should ever alter a file without user action.

We'd def take help on this.  Right now this is in the backlog and we don't have the resources to get around to this until a lot of current work items are worked through.  Thanks!

**Comment by smaillet** [comment]
@CyrusNajmabadi I'd be glad to help but I don't have a clue where to even look to find the source of this. As far as I can tell the GUI editor is part of VS itself. Which would make this bug in the wrong place, 
but the VS team moved it here from the VS community site...

[.editorconfig is applicable even if one doesn't even include support for building managed code.]

Thus without understanding WHAT the problem is or where the code for that exists it's impossible for anyone to help.

I'm confused that the team owning this is prioritizing "feature" work ahead of this as it makes the GUI editor completely useless. I, and I suspect ***many***, if not most, have disabled the editor and set a simple text editor as the default in VS.

**Comment by CyrusNajmabadi** [comment]
>  As far as I can tell the GUI editor is part of VS itself. 

I don't believe that is the case.  A former roslyn member wrote it and i think the code is in this repo.  

> I'm confused that the team owning this is prioritizing "feature" work ahead of this

When prioritizing, other things were deemed more important.  So this is currently in the backlog.  If that changes, it will move into a milestone to be done.  But currently that's not the case.

**Comment by smaillet** [comment]
>I don't believe that is the case. A former roslyn member wrote it and i think the code is in this repo.

And that would be where exactly? No-one can help without some knowledge of where.

**Comment by CyrusNajmabadi** [comment]
> And that would be where exactly? No-one can help without some knowledge of where.

I genuinely don't know.  I've never worked on it.  I did a search in roslyn for EditorConfig and found `Microsoft.VisualStudio.LanguageServices.EditorConfigSettings.SettingsEditorPane` which might be a reasonable place to start.

**Comment by smaillet** [comment]
>Microsoft.VisualStudio.LanguageServices.EditorConfigSettings.SettingsEditorPane which might be a reasonable place to start.

How did you get that? WHERE (FULL URL) is that located? I just did a GH search on that and found nothing.

It's maddening this has been open for so long and major releases of VS have occurred with this fundamentally broken UI that users have to disable for things to work right. (Not to mention how hard it is to get someone to definitively point to the source code for that so we can help.)

**Comment by JoeRobich** [comment]
> WHERE (FULL URL) is that located?

https://github.com/dotnet/roslyn/blob/main/src/VisualStudio/Core/Def/EditorConfigSettings/SettingsEditorPane.cs

> how hard it is to get someone to definitively point to the source code for that so we can help

This readme might help you get started https://github.com/dotnet/roslyn/blob/main/src/VisualStudio/Core/Def/EditorConfigSettings/readme.md

I think it would make sense to put a breakpoint on https://github.com/dotnet/roslyn/blob/6a135ea0ca8b7332d11fa78d68f0a14a65f2644f/src/VisualStudio/Core/Def/EditorConfigSettings/Common/EditorTextUpdater.cs#L34

**Comment by CyrusNajmabadi** [comment]
> How did you get that?

I did a ctrl-t and searched for EditorConfig.  The location of it on my system is: D:\github\roslyn\src\VisualStudio\Core\Def\EditorConfigSettings\SettingsEditorPane.cs

> Not to mention how hard it is to get someone to definitively point to the source code for that so we can help.

The challenge is that there likely isn't anyone who definitively knows.  I can try to help.  but i'm starting from the same position as you are here :)

**Comment by CyrusNajmabadi** [comment]
Note: due to the holidays coming, i recommend using https://discord.com/channels/143867839282020352/598678594750775301 to potentially communicate with peoplee over things in real-time.
